src/qibolab/pulses/sequence.py

```diff
@@ -34,8 +34,13 @@ def channel_duration(self, channel: str) -> float:
         """Duration of the given channel."""
         return sum(item.duration for item in self[channel])
 
-    def __add__(self, other: "PulseSequence") -> "PulseSequence":
-        """Create a PulseSequence which is self + necessary delays to
-        synchronize channels + other."""
-        # TODO: implement
-        ...
+    def append(self, other: "PulseSequence") -> None:
+        """Appends other in-place such that the result is self + necessary
+        delays to synchronize channels + other."""
+        tol = 1e-12
```

Comment:
Top-level constant?

Reply:
Well, it is used only here, I don't see any reason to move it to top level.

Reply:
Avoid hiding constants. If nothing else, it will be documented.

Reply:
It is not hidden, it is right there. I can mention it in the docstring though.

Reply:
What is your motivation to actively keep this outside top-level?

As a constant, not only it would be documented, but other functions could also make use of it (both outside Qibolab, inside Qibolab, or even for testing). In the docstring it would be only human-readable, and then possibly copied when needed.

Reply:
> What is your motivation to actively keep this outside top-level?

I don't see the point. A lot of things can be generalized, overengineered, and made future-proof or future-usable for a future that is not known/

>  but other functions could also make use of it 

This is exactly what I want to avoid - people accidentally using things that they shouldn't. If you want to introduce a global system for handling numerical imprecisions that is suitable for any context, maybe it can be discussed outside of this PR.

Reply:
I don't care specifically about numerical accuracy. Nor I want any specific use of this value.
I'm not proposing anything specific to this instance of the problem, but just a general guideline: avoid hard-coding values. Being them numbers or strings.

If you need values in your code that are not part of the user input, give them a name and document them properly.

Moreover, there is no way this request corresponds to overengineering: you would use the exact same thing (you're assigning a name anyhow), without any further feature. The only plain addition would be the explicit docs.
And yes, you could avoid documenting internals. But we decided not to do that (many people are working on the projects, so internals' documentation provides benefits for further contributors).